src/Battlescape/BattlescapeGame.cpp

```diff
@@ -1807,22 +1807,30 @@ void BattlescapeGame::primaryAction(Position pos)
 		}
 		else if ((_currentAction.type == BA_PANIC || _currentAction.type == BA_MINDCONTROL || _currentAction.type == BA_USE) && _currentAction.weapon->getRules()->getBattleType() == BT_PSIAMP)
 		{
-			if (_save->selectUnit(pos) && _save->selectUnit(pos)->getFaction() != _save->getSelectedUnit()->getFaction() && _save->selectUnit(pos)->getVisible())
-			{
-				_currentAction.updateTU();
-				_currentAction.target = pos;
-				if (!_currentAction.weapon->getRules()->isLOSRequired() ||
-					std::find(_currentAction.actor->getVisibleUnits()->begin(), _currentAction.actor->getVisibleUnits()->end(), _save->selectUnit(pos)) != _currentAction.actor->getVisibleUnits()->end())
-				{
-					// get the sound/animation started
-					getMap()->setCursorType(CT_NONE);
-					_parentState->getGame()->getCursor()->setVisible(false);
-					_currentAction.cameraPosition = getMap()->getCamera()->getMapOffset();
-					statePushBack(new PsiAttackBState(this, _currentAction));
-				}
-				else
+			if (_save->selectUnit(pos) && _save->selectUnit(pos)->getVisible())
+			
+				bool isFriendlyTargetingAllowed = _currentAction.weapon->getRules()->isFriendlyTargetingAllowed();
+				bool isEnemyTargetingAllowed = _currentAction.weapon->getRules()->isEnemyTargetingAllowed();
+				bool differentFactions = _save->selectUnit(pos)->getFaction() != _save->getSelectedUnit()->getFaction();
+				if ((isEnemyTargetingAllowed && differentFactions)
+					|| (isFriendlyTargetingAllowed && !differentFactions)
+					|| (isFriendlyTargetingAllowed && isEnemyTargetingAllowed))
```

Comment:
The third part of the OR condition in not needed, correct? Or does it have some hidden meaning?

Reply:
No, you are right.